pkg/service/catalog/api/types.go

```diff
@@ -0,0 +1,21 @@
+package api
+
+import (
+	kapi "k8s.io/kubernetes/pkg/api"
+)
+
+type CatalogService struct {
+	// Target is a reference to a service
```

Comment:
If you are a top level object you have typemeta and objectmeta.

On Oct 2, 2015, at 10:16 AM, Andy Goldstein <notifications@github.com>
wrote:

In pkg/service/catalog/api/types.go
<https://github.com/openshift/origin/pull/4822#discussion_r41025704>:

> @@ -0,0 +1,21 @@
> +package api
> +
> +import (
> +	kapi "k8s.io/kubernetes/pkg/api"
> +)
> +
> +type CatalogService struct {
> +	// Target is a reference to a service

You'd oc get catalogservices/foo to get all the services in catalog foo. I
don't anticipate needing to retrieve one by name, but maybe? We will want
to filter based on labels, so @ingvagabund <https://github.com/ingvagabund>
go ahead and add ObjectMeta.

I guess let's add TypeMeta too. It can't hurt anything.

—
Reply to this email directly or view it on GitHub
<https://github.com/openshift/origin/pull/4822/files#r41025704>.